course/page/upload.py

```diff
@@ -70,9 +70,21 @@ def clean_uploaded_file(self):
                     % {"allowedsize": filesizeformat(self.max_file_size),
                         "uploadedsize": filesizeformat(uploaded_file.size)})
 
-        if self.mime_types is not None and self.mime_types == ["application/pdf"]:
-            if uploaded_file.read()[:4] != b"%PDF":
-                raise forms.ValidationError(_("Uploaded file is not a PDF."))
+        if self.mime_types is not None:
+            if self.mime_types == ["application/pdf"]:
+                if uploaded_file.read()[:4] != b"%PDF":
+                    raise forms.ValidationError(_("Uploaded file is not a PDF."))
+            if self.mime_types == ["application/x-ipynb+json"]:
+                try:
+                    # make sure it is loadable json
+                    import json
+                    data = json.load(uploaded_file)
+
+                    # check for a notebook format of at least 4
+                    assert int(data["nbformat"]) >= 4
```

Comment:
Don't use `assert` for input checking. (might be inadvertently disabled via `-O`)

Reply:
good point.  added a new raise, but take a look

Reply:
Is that valid syntax? What does it raise?

Reply:
it is probably better to raise a specific type of error.  added a `ValueError`